lib/shelley/src/Cardano/Wallet/Shelley/Transaction.hs

```diff
@@ -15,92 +14,166 @@
+import qualified Cardano.Crypto.Wallet as CC
+import qualified Crypto.PubKey.Ed25519 as Ed25519
 import qualified Data.ByteString as BS
-
+import qualified Data.Map as Map
+import qualified Data.Set as Set
+import qualified Shelley.Spec.Ledger.BaseTypes as SL
+import qualified Shelley.Spec.Ledger.Keys as SL
+import qualified Shelley.Spec.Ledger.Tx as SL
+import qualified Shelley.Spec.Ledger.UTxO as SL
 
 newTransactionLayer
     :: forall (n :: NetworkDiscriminant) k t.
-        ( t ~ IO Shelley)
+        ( t ~ IO Shelley
+        , WalletKey k
+        )
     => Proxy n
     -> ProtocolMagic
+    -> EpochLength
     -> TransactionLayer t k
-newTransactionLayer _proxy _protocolMagic = TransactionLayer
-    { mkStdTx = notImplemented "mkStdTx"
+newTransactionLayer _proxy _protocolMagic epochLength = TransactionLayer
+    { mkStdTx = _mkStdTx
     , mkDelegationJoinTx = notImplemented "mkDelegationJoinTx"
     , mkDelegationQuitTx = notImplemented "mkDelegationQuitTx"
     , decodeSignedTx = notImplemented "decodeSignedTx"
-    , estimateSize = notImplemented "estimateSize"
-    , estimateMaxNumberOfInputs = notImplemented "estimateMaxNumberOfInputs"
-    , validateSelection = notImplemented "validateSelection"
-    , allowUnbalancedTx = notImplemented "allowUnbalancedTx"
-    }
-
--- | Construct a ("fake") genesis block from genesis transaction outputs.
---
--- The genesis data on haskell nodes is not a block at all, unlike the block0 on
--- jormungandr. This function is a method to deal with the discrepancy.
-genesisBlockFromTxOuts :: BlockchainParameters -> [TxOut] -> Block
-genesisBlockFromTxOuts bp outs = Block
-    { delegations  = []
-    , header = BlockHeader
-        { slotId =
-            SlotId 0 0
-        , blockHeight =
-            Quantity 0
-        , headerHash =
-            coerce $ getGenesisBlockHash bp
-        , parentHeaderHash =
-            Hash (BS.replicate 32 0)
-        }
-    , transactions = mkTx <$> outs
+    , estimateSize = _estimateSize
+    , estimateMaxNumberOfInputs = _estimateMaxNumberOfInputs
+    , validateSelection = const $ return ()
+    , allowUnbalancedTx = True
     }
   where
-    mkTx out@(TxOut (Address bytes) _) =
-        Tx (Hash $ blake2b256 bytes) [] [out]
+    _mkStdTx
+        :: (Address -> Maybe (k 'AddressK XPrv, Passphrase "encryption"))
+        -> SlotId -- ^ The current slot
+        -> [(TxIn, TxOut)]
+        -> [TxOut]
+        -> Either ErrMkTx (Tx, SealedTx)
+    _mkStdTx keyFrom slot ownedIns outs = do
+        let ins' = map (toCardanoTxIn . fst) ownedIns
+        let outs' = map toCardanoTxOut outs
+        let fee = Coin $ sum (map (getCoin . coin . snd) ownedIns)
+                - sum (map (getCoin . coin) outs)
+        let certs = []
+
+        -- TODO: The SlotId-SlotNo conversion based on epoch length would not
+        -- work if the epoch length changed in a hard fork.
+        let timeToLive = (toSlotNo epochLength slot) + 7200
```

Comment:
7200 ? Avoid magic numbers, make it a named constant and explain the rationale of choosing `7200` (ultimately, we want that configurable via the API). Can you make sure to let Alex know about this?